nucypher/policy/maps.py

```diff
@@ -35,12 +35,63 @@
 from nucypher.crypto.kits import UmbralMessageKit
 from nucypher.crypto.powers import DecryptingPower, SigningPower
 from nucypher.crypto.signing import SignatureStamp
-from nucypher.crypto.splitters import signature_splitter
-from nucypher.crypto.umbral_adapter import KeyFrag, PublicKey, Signature
+from nucypher.crypto.splitters import signature_splitter, hrac_splitter, kfrag_splitter
+from nucypher.crypto.umbral_adapter import KeyFrag, VerifiedKeyFrag, PublicKey, Signature
 from nucypher.crypto.utils import keccak_digest, encrypt_and_sign, verify_eip_191
 from nucypher.network.middleware import RestMiddleware
 
 
+class AuthorizedKeyFrag:
+
+    _splitter = BytestringSplitter(
+        hrac_splitter, # HRAC
+        BytestringSplitter((bytes, WRIT_CHECKSUM_SIZE)), # kfrag checksum
+        signature_splitter, # alice signature
+        kfrag_splitter,
+        )
+
+    @staticmethod
+    def _kfrag_checksum(kfrag: Union[KeyFrag, VerifiedKeyFrag]) -> bytes:
+        # TODO: assuming here that KeyFrag and VerifiedKeyFrag have the same byte representation.
+        # It's a little hacky... Should we have `verified_kfrag.unverify()` or something?
+        return keccak_digest(bytes(kfrag))[:WRIT_CHECKSUM_SIZE]
+
+    @classmethod
+    def create(cls, hrac: bytes, kfrag: VerifiedKeyFrag, alice_stamp: SignatureStamp) -> 'AuthorizedKeyFrag':
```

Comment:
Perhaps `from_verified_kfrag` adds a bit more semantic density here.

Reply:
Yes, that's a good variant. I didn't particularly like the name myself, it was just a placeholder. Another possibility is `construct_by_alice`.

Reply:
Or rather `construct_by_publisher`, I guess

Reply:
I like `construct_by_publisher` too 👍🏻

Reply:
Renamed, along with remaining "alice" usage in `maps.py`